Gemfile.lock

```diff
@@ -94,6 +94,7 @@ PATH
     avs (0.1.0)
     check_in (0.1.0)
     claims_api (0.0.1)
+      dry-schema
```

Comment:
do we need this? can we not get away with using an existing tooling or potentially creating a strong params type check in the forefront and not deeply nested in the call to BGS itself? seems like we should validate this well beforehand.

Reply:
The validation happens in the close to the first line in the controller. Regarding reusing the existing tool, I was curious about that too ([my comment above](https://github.com/department-of-veterans-affairs/vets-api/pull/16661/files#diff-26c32ebd6e89acace52097bb6aed4ddd261b2fec40fc7e19b4ff8f56b7f0e042R14))

Reply:
ahh, the link wasn't working but I think you're referring to the question about json_schemer, which yes, I believe you should be able to achieve the same thing with that gem and not have to bring this in. it's also interesting how we have to load the tooling to provide the error messages (at least that's what [I'm understanding from this](https://github.com/department-of-veterans-affairs/vets-api/pull/16661#discussion_r1597279323)). Plus those hints aren't even being utilize yet from what I can tell - maybe this is incorrect? - so I'm not sure why we have lines bringing them in for this PR

Reply:
The hints can be seen in the request spec. It's what gives the full set of validation error messages for each property [like so](https://github.com/department-of-veterans-affairs/vets-api/blob/dash/oren/API-34439/read-poa-request-pagination/modules/claims_api/spec/requests/v2/power_of_attorney_requests/index/request_spec.rb#L102-L103).

The [example](https://dry-rb.org/gems/dry-schema/1.13/extensions/hints/) in their docs shows how `messages` combines the base `errors` + the `hints`. So in this example the base error was how the value wasn't "filled" but then also shows the hint about how once it is filled it also needs to be "greater than 18":
```ruby
result = schema.call(email: 'jane@doe.org', age: '')
result.messages.to_h
# {:age=>["must be filled", "must be greater than 18"]}
```